include/midi_pedal.hpp

```diff
@@ -7,7 +7,7 @@ typedef void (*note_off_callback_t)(uint8_t, uint8_t);
 class MidiPedalAdapter {
   public:
 	MidiPedalAdapter(note_on_callback_t note_on_callback, note_off_callback_t note_off_callback)
-	    : held_lo{0}, held_hi{0}, pedal_down(false), note_on_callback(note_on_callback),
+	    : held_lo{0}, held_hi{0}, pedal_down{0}, note_on_callback(note_on_callback),
 	      note_off_callback(note_off_callback) {}
 
 	void set_pedal(uint8_t channel, bool pedal_down) {
```

Comment:
in line 14, you forgot one occurrence of changing `this->pedal_down` to `this_pedal_down[channel]`